lib/spack/spack/test/cmd/env.py

```diff
@@ -1948,6 +1948,154 @@ def test_view_link_roots(tmpdir, mock_fetch, mock_packages, mock_archive,
                                  (spec.version, spec.compiler.name)))
 
 
+def test_view_link_type_symlink(tmpdir, mock_fetch, mock_packages, mock_archive,
```

Comment:
We could make all of these tests simpler by:

1. Removing the definitions from the environment
2. Setting the environment specs to `[mpileaks]`
3. Removing the projections, select, exclude, and link keys from the view descriptor
4. Adjusting the paths we check for appropriately

We should also be able to parametrize the three tests into one.

```
@pytest.mark.parametrize('type', ['symlink', 'hardlink', 'copy'])
def test_view_link_type(type, tmpdir, mock_fetch, mock_packages, mock_archive,
...
...
assert os.path.islink(...) == (type == 'symlink')
```